### PR TITLE
PIM-4007: Rewrite the mass edit family (attribute requirements)

### PR DESCRIPTION
**Pull request by grena**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | Y
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-4007
| DB schema updated?   | N
| Migration script?    | -
| Doc PR               | -

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Reader/MassEdit/FilteredFamilyReader.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     * @param array $filters
+     *
+     * @return \Doctrine\Common\Collections\ArrayCollection
+     */
+    protected function getFamilies(array $filters)
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setRequired(['field', 'operator', 'value']);
+
+        // In the case of the Family reader, we only have 1 filter for IDS
```
Comment (solivier) [inline src/Pim/Bundle/EnrichBundle/Reader/MassEdit/FilteredFamilyReader.php]: comments = confession of failure

File: spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/SetAttributeRequirementsSpec.php [hunk 2]
```diff
@@ -94,4 +94,62 @@ function it_initializes_attribute_requirements_with_all_channels_and_attributes_
...
     }
+
+    function it_returns_well_formatted_actions_for_batch_job(
+        AttributeInterface $attr1,
+        AttributeInterface $attr2,
+        ChannelInterface $channel1,
+        ChannelInterface $channel2,
+        AttributeRequirementInterface $attrRequirement1,
+        AttributeRequirementInterface $attrRequirement2,
+        AttributeRequirementInterface $attrRequirement3
```
Comment (willy-ahva) [inline spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/SetAttributeRequirementsSpec.php]: change 1, 2 ,3 by real name pls. To be more understandable

File: spec/Pim/Bundle/EnrichBundle/Processor/MassEdit/SetAttributeRequirementsSpec.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        StepExecution $stepExecution,
+        ValidatorInterface $validator,
+        FamilyInterface $family,
+        JobExecution $jobExecution,
+        MassEditJobConfiguration $massEditJobConf,
+        AttributeInterface $attribute,
+        ChannelInterface $channel1,
+        ChannelInterface $channel2,
+        AttributeRequirementInterface $attrReq1,
+        AttributeRequirementInterface $attrReq2
```
Comment (willy-ahva) [inline spec/Pim/Bundle/EnrichBundle/Processor/MassEdit/SetAttributeRequirementsSpec.php]: same here for naming

File: spec/Pim/Bundle/EnrichBundle/Reader/MassEdit/FilteredFamilyReaderSpec.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    function it_reads_families(
+        $massEditRepository,
+        $familyRepository,
+        StepExecution $stepExecution,
+        JobExecution $jobExecution,
+        MassEditJobConfiguration $massEditJobConf,
+        FamilyInterface $family1,
+        FamilyInterface $family2
```
Comment (willy-ahva) [inline spec/Pim/Bundle/EnrichBundle/Reader/MassEdit/FilteredFamilyReaderSpec.php]: naming

**Pull request merged**